jerry-main/main-unix-minimal.c

```diff
@@ -0,0 +1,128 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string.h>
+
+#include "jerry-api.h"
+#include "jerry-port-default.h"
+
+/**
+ * Maximum size of source code / snapshots buffer
+ */
+#define JERRY_BUFFER_SIZE (1048576)
+
+/**
+ * Standalone Jerry exit codes
+ */
+#define JERRY_STANDALONE_EXIT_CODE_OK   (0)
+#define JERRY_STANDALONE_EXIT_CODE_FAIL (1)
+
+static uint8_t buffer[ JERRY_BUFFER_SIZE ];
+
+static const uint8_t *
+read_file (const char *file_name,
+           size_t *out_size_p)
+{
+  FILE *file = fopen (file_name, "r");
+  if (file == NULL)
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: failed to open file: %s\n", file_name);
+    return NULL;
+  }
+
+  size_t bytes_read = fread (buffer, 1u, sizeof (buffer), file);
+  if (!bytes_read)
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: failed to read file: %s\n", file_name);
+    fclose (file);
+    return NULL;
+  }
+
+  fclose (file);
+
+  *out_size_p = bytes_read;
+  return (const uint8_t *) buffer;
+} /* read_file */
+
+static void
+print_help (char *name)
+{
+  jerry_port_console ("Usage: %s [OPTION]... [FILE]...\n"
+                      "\n"
+                      "Options:\n"
+                      "  -h, --help\n"
+                      "\n",
+                      name);
+} /* print_help */
+
+int
+main (int argc,
+      char **argv)
+{
+  if (argc <= 1 || (argc == 2 && (!strcmp ("-h", argv[1]) || !strcmp ("--help", argv[1]))))
+  {
+    print_help (argv[0]);
+    return JERRY_STANDALONE_EXIT_CODE_OK;
+  }
+
+  jerry_init (JERRY_INIT_EMPTY);
+  jerry_value_t ret_value = jerry_create_undefined ();
+
+  const char *file_name;
+  int i = 1;
+
+  for (; i < argc; i++)
```

Comment:
`i` is not used outside the `for`, so please combine these lines.

`for (int i = 1; i < argc; i++)`